nuxeo-services/nuxeo-platform-filemanager-core-listener/src/main/java/org/nuxeo/ecm/platform/filemanager/core/listener/MimetypeIconUpdater.java

```diff
@@ -132,12 +138,17 @@ public void updateBlobProperty(DocumentModel doc, MimetypeRegistry mimetypeServi
         }
 
         Blob blob = dirtyProperty.getValue(Blob.class);
-        if (blob != null && (blob.getMimeType() == null || blob.getMimeType().equals(OCTET_STREAM_MT))) {
-            // update the mimetype (if not set) using the the mimetype registry
-            // service
-            blob = mimetypeService.updateMimetype(blob);
-            doc.setPropertyValue(fieldPath, (Serializable) blob);
+        if (blob != null) {
```

Comment:
Could use an early exit to gain indentation:
```
if (blob == null) {
    return;
}
```

Reply:
yes i see, i should say that i `blob !=null` is my preferred one, we will gain on lines of code and i think is more clear . but as we discuss the last time and to be consistent with other code in nuxeo. i will replace it the early exit

Reply:
Thanks. It should probably be in the dev guidelines, but usually we prefer early return because:
- It reduces cyclomatic complexity.
- It reduces indentation.
- It makes the code more readable, handling the exception/null/easy cases first then implementing the pure logic.